File: CS50/sentimental-mario-more/mario.py
from cs50 import get_int


def hash(ctype, times):
    print(ctype * times, end="")


def main():
    while True:
        input = get_int("Height: ")

        if (input >= 1 and input <= 8):
            break

    hash_index = 1
    for i in range(input):
        hash(" ", input - 1 - i)
        hash("#", hash_index + i)
        hash(" ", 2)
        hash("#", hash_index + i)
        print()

if __name__ == "__main__":
    main()
    

File: CS50/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int input; // user input
    int space; // gaps towards hashes
    int row; // rows (iterator)
    int col; // columns (iterator)
    do
    {
        input = get_int("Height: "); // user inputs height of the pyramid
    }
    while (!((input <= 8) && (input >= 1))); // input condition, input should be between 1-8, cannot input chars

    for (row = 0; row < input; row++) // iterator for numbers of row according to user input
    {
        for (space = 0; space < input - row - 1; space++) // iterator for number of columns, example: user input is 4 so loop will print spaces untill space < 3 (4 - 0 - 1); it prints 3 spaces
        {
            printf(" ");
        }
        for (col = 0; col <= row; col++) // loop printing first half of hashes
        {
            printf("#");
        }
        printf("  "); // gap between hashes is constant, it can be stored as a string of spaces, which will be printed in each row and each iteration of loop
        for (col = 0; col <= row; col++) // loop printing second half of hasesh
        {
            printf("#");
        }
        printf("\n"); // statement printing new line after whole iteration of loop
    }
}